### ISSUE TITLE
[Documentation] Learning rack with examples

### ISSUE THREAD

**Issue by rubyFeedback** [rack/rack#1760]
Hey guys,

I have been using sinatra for quite some time now; sinatra is quite ok, but I feel quite limited.
For example, I can not easily use more than one image directory. The images I have on my
local system are spread all over everywhere. There are some workarounds, including just
using base64 :P but I am getting tired of workarounds - so I decided I'll abandon sinatra in
the long run (also because the code quality is not good and there are barely any comments
in sinatra). So now I am digging into rack!

I found useful tutorials and I am learning that, but right now I can not even get img src=""
to work, even when using 'Content-Type' => 'text/html'.

I am sure this is a user error, but right now I can not figure it out.

So I came here hoping there would be many small rack-specific examples, but looking
at the examples, I only see the epic lobster. But that's it.

I am not sure whether you guys want to distribute more examples as-is, but perhaps you
could link in to a tutorial that really shows TONS of simple stuff? Including hello world
and how to show an image? Yes, this is super-noob material but right now I am a noob
too. And I'd love to have more examples reading and learning rack. Right now I just
gather what is available on the www, StackOverflow and what not ... in short, I go
to external sites to learn more about rack, which is ... weird. :P

So to summarize this: please show a simple example how to show an image in the
same directory using only rack (but content type html). I'd re-word this more generally
though, more examples would be REALLY great. I don't know why but ruby suffers
immensely from people hating to use examples. It's weird when StackOverflow has
better examples than official libraries in general, but that's an aside - at the least
rack is still better documented than about 99% of the other ruby-related gems out
there ... :P

**Comment by jeremyevans** [comment]
We appreciate the feedback. However, I don't think we want to give more examples of how to develop web applications directly on top of Rack.  Rack is designed to be a layer to connect web applications (generally written in a web framework) to web servers. Rack is conceptually quite simple, and you can develop a web application directly on top of it without a framework, but you have to know what you are doing.  I don't think developing typical web applications directly on top of Rack is something that any of the committers would recommend, and I don't think there are plans to develop the documentation you are requesting.  That may be frustrating for you, and I apologize for that.
